### PR TITLE
chore: add AriaSnapshot internal type

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/server/injected/injectedScript.ts [hunk 1]
```diff
@@ -215,10 +216,27 @@ export class InjectedScript {
     return new Set<Element>(result.map(r => r.element));
   }
 
-  ariaSnapshot(node: Node, options?: { mode?: 'raw' | 'regex' }): string {
+  ariaSnapshot(node: Node, options?: { mode?: 'raw' | 'regex', id?: boolean }): string {
```
Comment (yury-s) [inline packages/playwright-core/src/server/injected/injectedScript.ts:219]: id -> ids?: Map... ?

File: packages/playwright-core/src/server/injected/injectedScript.ts [hunk 2]
```diff
@@ -215,10 +216,27 @@ export class InjectedScript {
...
+
+  ariaSnapshotAsObject(node: Node): AriaSnapshot {
+    return generateAriaTree(node as Element);
+  }
+
+  ariaSnapshotElement(snapshot: AriaSnapshot, elementId: number): Element | null {
+    return snapshot.elements.get(elementId) || null;
+  }
+
+  renderAriaTree(ariaNode: AriaNode, options?: { mode?: 'raw' | 'regex', id?: boolean}): string {
```
Comment (yury-s) [inline packages/playwright-core/src/server/injected/injectedScript.ts:234]: what is id?: boolean?

File: packages/playwright-core/src/server/recorder/chat.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  clearHistory() {
+    this._history = [];
+  }
+
+  async post<T>(prompt: string): Promise<T | null> {
+    await this._append('user', prompt);
+    let text = await asString(await this._post());
+    if (text.startsWith('```json') && text.endsWith('```'))
+      text = text.substring('```json'.length, text.length - '```'.length);
+    for (let i = 0; i < 3; ++i) {
```
Comment (yury-s) [inline packages/playwright-core/src/server/recorder/chat.ts:44]: why in the loop?

**Review by yury-s — APPROVED** [review]

File: packages/playwright-core/src/server/recorder/chat.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/recorder/chat.ts:44]: To retry :)

**Pull request merged**